Tests/BackendIntegrationTests/StoreKitObserverModeIntegrationTests.swift

```diff
@@ -81,13 +81,21 @@ class StoreKit2ObserverModeIntegrationTests: StoreKit1ObserverModeIntegrationTes
 
 }
 
+class StoreKit2JWSObserverModeIntegrationTests: StoreKit2ObserverModeIntegrationTests {
+    override var usesStoreKit2JWS: Bool { return true }
+}
+
 class StoreKit1ObserverModeIntegrationTests: BaseStoreKitObserverModeIntegrationTests {
 
     override class var storeKit2Setting: StoreKit2Setting { return .disabled }
 
     func testPurchaseOutsideTheAppPostsReceipt() async throws {
         try self.testSession.buyProduct(productIdentifier: Self.monthlyNoIntroProductID)
 
+        // In JWS mode, transaction takes a bit longer to be processed after `buyProduct`
+        // We need to wait so `restorePurchases` actually posts it.
+        try? await Task.sleep(nanoseconds: DispatchTimeInterval.seconds(2).nanoseconds)
```

Comment:
Ditto, let's avoid these sleeps. Let's replace it with an explicit wait for whatever we need (See `StoreKitTestHelpers` for examples).